### PR TITLE
refactor: cli support preview and preview apply

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
Need cli support preview and preview apply

**Related Issue:**
#2181

### CODE REVIEW

File: staging/utils/json/jsonpatch.go [hunk 1]
```diff
@@ -1,11 +1,62 @@
...
+		for _, sf := range skipFields {
+			field := sf
+			if toCamel {
+				field = strs.CamelizeDownFirst(field)
+			}
+
+			ope := op.Get(field)
+			ppe := pp.Get(field)
+			if ope.Exists() && ppe.Exists() {
+				ob, err = sjson.DeleteBytes(ob, field)
```
Comment (alexcodelf) [inline staging/utils/json/jsonpatch.go:39]: Can you explain the reason that we need delete the field here?

File: pkg/apis/resource/basic_view.go [hunk 2]
```diff
@@ -33,12 +33,18 @@ import (
 	"github.com/seal-io/walrus/utils/validation"
 )
 
+type ResourceRunInfo struct {
+	Draft         bool              `json:"draft,default=false"`
```
Comment (alexcodelf) [inline pkg/apis/resource/basic_view.go:37]: Draft controls the behavior of resource, shall we put it into resource run info?

**Review by gitlawr — DISMISSED** [review]
LGTM

File: pkg/cli/cmd/preview.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		resultChan <- r.Success
+		return nil
+	})
+
+	return wg.Wait()
+}
+
+func previewExample(action string) string {
+	return fmt.Sprintf(`  # Generate preview of the configuration in the walrus-file.yaml 
+  $ walrus %s -f walrus-file.yaml --label key=value
```
Comment (alexcodelf) [inline pkg/cli/cmd/preview.go:111]: should it be labels?

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**